src/Components/VisualObjectInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Gui\Components;
+
+interface VisualObjectInterface
+{
+    public function getAutoSize();
```

Comment:
as it is an interface, you can just put the annotations over here, and in the implementation you just use @inheritdoc for all methods.